### PR TITLE
Hit-API

### PR DESCRIPTION
**Pull request by ibr5500**
### In this project I have:

> - Create a new game.
> - Create a function that displays the scores of the players when clicking on the refresh button.
> - Create a function that adds new scores when clicking on submit button.

### CODE REVIEW

**Comment by ibr5500** [comment]
I have a problem in the deployed version the user names and the scores show undefined. But in the undeployed version, I have fixed it. However, it is still not updated to the deployed one.

File: src/modules/usersScore.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    playersData = [];
+
+    url = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/QE8nBkXmdfd1YoxrlAjT/scores/';
+
+    getData = () => {
+      const scores = document.getElementById('scores-container');
+      scores.innerHTML = this.playersData.map((elem, index) => `<p class=${index % 2 !== 0 ? 'row-bg' : ''} >${elem.user}: <span>${elem.score}</span></p>`).join('');
+    }
+
```
Comment (uwadonat) [inline src/modules/usersScore.js:15]: - [x] When I Click on refresh button the score should be updated with new values only. Now this action cause duplication of same values.Please fix this.
Screenshot from 2022-05-19 13-42-38

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **[ibr5500](https://github.com/ibr5500),** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your commit  and pull request messages are very descriptive ✔️
- You are following JavaScript best practices ✔️
almost


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/usersScore.js (hunk 1 above)
Comment (ibr5500) [reply src/modules/usersScore.js:15]: Hi @uwadonat, 
Do you have any suggestions to fix that?

File: src/index.js [hunk 2]
```diff
@@ -1 +1,18 @@
...
+  event.preventDefault();
+  const user = form.name.value;
+  const score = form.score.value;
+  newUserScore.addNewScore({ user, score });
+  form.name.value = '';
+  form.score.value = '';
+});
+
+const refersh = document.getElementById('refresh');
+refersh.addEventListener('click', newUserScore.fetchData);
```
Comment (omar-muhamad) [inline src/index.js]: - [x] Please fix the double render of your data as the last reviewer mentioned.
You can do it by passing a custom function to refresh the button event listener and in it you can reset the data in the dom before you call `fetchData`.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ibr5500,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/index.js (hunk 2 above)
Comment (ibr5500) [reply src/index.js]: Ok, I will try to do that. Thank you for the suggestion.

Comment (ibr5500) [reply src/index.js:17]: Hi @Omar-Muhamad, I tried this but it's still not working 
`refersh.addEventListener('click', () => {
    document.getElementById('scores-container').innerHTML = "";
    return newUserScore.fetchData;
});`

Comment (ibr5500) [reply src/index.js]: Hello @Omar-Muhamad,
I have solved the problem.
Thank you for the code review.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @ibr5500,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**